### DISCUSSION TITLE
Option to change units for wireframes

### DISCUSSION THREAD

**Discussion by goatsneez** [ortk95/planetmapper#332] [Ideas]
It seems that the current units are either (rad/dec, km, arcsec) for plotting the wireframes. What would be the most pragmatic way to scale arcsec-> deg or arc-min? Would it be possible to implement scalar argument into these function calls to be able to easily scale them to desired units; e.g. from km -> Rjup , arcsec->deg or arc-min?

**Comment by ortk95** [comment]
> [!NOTE]
> See https://github.com/ortk95/planetmapper/discussions/332#discussioncomment-8494798 for details of the new `scale_factor` argument in PlanetMapper 1.9.1 and above _(note added 2024-09-19)._

There isn't currently a way to do that natively with PlanetMapper, but it would definitely be really useful, so I'll add it to the to-do list!

In the meantime, it should be relatively straightforward to do this yourself. All the wireframes are implemented using the (private) method `Body._plot_wireframe` with a couple of custom arguments to define the projection - e.g. this is the code that generates the `km` wireframe: https://github.com/ortk95/planetmapper/blob/460b3762fa7cd8eeaa3d3583acfe2cb930a40f91/planetmapper/body.py#L2634-L2644

You should therefore be able to create a projection with custom scaling by calling `_plot_wireframe` with a custom value for `coordinate_func`. For example, to create a wireframe with units of `r_jup`, you can do something like:

```python
def coordinate_func(ra, dec):
    x, y = body.radec2km(ra, dec)
    return x / body.r_eq, y / body.r_eq


body._plot_wireframe(
    coordinate_func=coordinate_func,
    transform=None,
    aspect_adjustable='datalim',
)
```

Obviously using an private method (like `_plot_wireframe`) is normally not recommended, as it may change in future versions of PlanetMapper, but in this case it should be a decent workaround until I get this implemented properly in the next release.

**Reply by ortk95** [reply]
Issue tracking the implementation of the new feature: #333

**Comment by ortk95** [comment]
In the latest release of PlanetMapper (v1.9.1), I've added a new `scale_factor` argument to allow the units of wireframe plots to be easily customised. For example:

```python
body.plot_wireframe_radec() # units of degrees
body.plot_wireframe_radec(scale_factor=3.14159/180) # units of radians

body.plot_wireframe_km() # units of km
body.plot_wireframe_km(scale_factor=1000) # units of m
body.plot_wireframe_km(scale_factor=1/body.r_eq) # units of planet radii

body.plot_wireframe_angular() # units of arcseconds
body.plot_wireframe_angular(scale_factor=1/60) # units of arcminutes
body.plot_wireframe_angular(scale_factor=1/3600) # units of degrees
```

See https://planetmapper.readthedocs.io/en/latest/documentation.html#planetmapper.Body.plot_wireframe_radec for more details.
